yt_dlp/extractor/steam.py

```diff
@@ -127,3 +127,44 @@ def _real_extract(self, url):
             raise ExtractorError('Could not find any videos')
 
         return self.playlist_result(entries, playlist_id, playlist_title)
+
+
+class SteamCommunityBroadcastIE(InfoExtractor):
+    _VALID_URL = r'https?://steamcommunity\.(?:com)/broadcast/watch/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://steamcommunity.com/broadcast/watch/76561199037072858',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        json_data = self._download_json(
+            'https://steamcommunity.com/broadcast/getbroadcastmpd/',
+            video_id, query={'steamid': f'{video_id}'}
+        )
+        formats = []
+        mpd_formats, mpd_subs = self._extract_mpd_formats_and_subtitles(json_data['url'], video_id)
+        format_, subs = self._extract_m3u8_formats_and_subtitles(json_data['hls_url'], video_id)
+
+        formats.extend(format_)
+        formats.extend(mpd_formats)
```

Comment:
```suggestion
        formats, subtitles = self._extract_mpd_formats_and_subtitles(json_data['url'], video_id)

        format_, subs = self._extract_m3u8_formats_and_subtitles(json_data['hls_url'], video_id)
        formats.extend(format_)
        self._merge_subtitles(subs, target=subtitles)
```
you also forgot to return the subs

Reply:
btw, did you check if yt-dlp can actually download the formats? There are some issues when downloading live dash

Reply:
I did the test for hls and  it goes well but i didn't test the dash video. After i test with dash video, it makes `HTTP 404 error` for dash video, should i commented code the refer to `mpd` format ?

Reply:
Yes